object-api-multipart.go

```diff
@@ -109,14 +110,16 @@ func (o objectAPI) ListMultipartUploads(bucket, prefix, keyMarker, uploadIDMarke
 		recursive = false
 	}
 	result.IsTruncated = true
+	result.MaxUploads = maxUploads
 	newMaxUploads := 0
-	prefixPath := path.Join(bucket, prefix)
-	if strings.HasSuffix(prefix, slashPathSeparator) {
-		// Add back the slash separator removed after 'path.Join'.
-		prefixPath = prefixPath + slashPathSeparator
-	}
+	// not using path.Join() as it strips off the trailing '/'.
+	// Also bucket should always be followed by '/' even if prefix is empty.
+	prefixPath := bucket + string(os.PathSeparator) + prefix
```

Comment:
Also why not do this inside a func pathJoin() like urlJoinPath on 'mc' which simplifies and retains trailing 'slashes' ?